apps/web/modules/auth/invite/page.tsx

```diff
@@ -0,0 +1,131 @@
+import { deleteInvite, getInvite } from "@/modules/auth/invite/lib/invite";
+import { createTeamMembership } from "@/modules/auth/invite/lib/team";
+import { authOptions } from "@/modules/auth/lib/authOptions";
+import { sendInviteAcceptedEmail } from "@/modules/email";
+import { Button } from "@/modules/ui/components/button";
+import { getServerSession } from "next-auth";
+import { getTranslations } from "next-intl/server";
+import Link from "next/link";
+import { after } from "next/server";
+import { DEFAULT_LOCALE, WEBAPP_URL } from "@formbricks/lib/constants";
+import { verifyInviteToken } from "@formbricks/lib/jwt";
+import { createMembership } from "@formbricks/lib/membership/service";
+import { getUser, updateUser } from "@formbricks/lib/user/service";
+import { ContentLayout } from "./components/content-layout";
+
+export const InvitePage = async (props) => {
+  const searchParams = await props.searchParams;
+  const t = await getTranslations();
+  const session = await getServerSession(authOptions);
+  const user = session?.user.id ? await getUser(session.user.id) : null;
+
+  try {
+    const { inviteId, email } = verifyInviteToken(searchParams.token);
+
+    const invite = await getInvite(inviteId);
+
+    if (!invite) {
+      return (
+        <ContentLayout
+          headline={t("auth.invite.invite_not_found")}
+          description={t("auth.invite.invite_not_found_description")}
+        />
+      );
+    }
+
+    const isInviteExpired = new Date(invite.expiresAt) < new Date();
+
+    const createMembershipAction = async () => {
+      "use server";
+
+      if (!session || !user) return;
+
+      await createMembership(invite.organizationId, session.user.id, {
+        accepted: true,
+        role: invite.role,
+      });
+      if (invite.teamIds) {
+        await createTeamMembership(invite, user.id);
+      }
+      await deleteInvite(inviteId);
+      await sendInviteAcceptedEmail(
+        invite.creator.name ?? "",
+        user?.name ?? "",
+        invite.creator.email,
+        user?.locale ?? DEFAULT_LOCALE
+      );
+      await updateUser(session.user.id, {
+        notificationSettings: {
+          ...user.notificationSettings,
+          alert: user.notificationSettings.alert ?? {},
+          weeklySummary: user.notificationSettings.weeklySummary ?? {},
+          unsubscribedOrganizationIds: Array.from(
+            new Set([
+              ...(user.notificationSettings?.unsubscribedOrganizationIds || []),
+              invite.organizationId,
+            ])
+          ),
+        },
+      });
+    };
+
+    if (isInviteExpired) {
+      return (
+        <ContentLayout
+          headline={t("auth.invite.invite_expired")}
+          description={t("auth.invite.invite_expired_description")}
+        />
+      );
+    } else if (!session) {
```

Comment:
I think there is no need for the `else if` here since we return in every condition, we can just use return after every condition with an `if`